V3.0 release dragonbones骨骼动画资源及组件 (#1522)
cocos-creator/creator-docs: en/editor/components/dragonbones.md

# Component Reference for DragonBones ArmatureDisplay

The ArmatureDisplay component can render and play DragonBones assets.

![dragonbones](./dragonbones/properties.png)

In the **Hierarchy** panel, select the node where you want to add the ArmatureDisplay component, and then click the **Add Components -> Components -> ArmatureDisplay** button under the **Inspector** panel to add the ArmatureDisplay component to the node.

- Please refer to the [DragonBones test case](https://github.com/cocos-creator/test-cases-3d/tree/v3.0/assets/cases/dragonbones) in [example-cases](https://github.com/cocos-creator/example-cases) for the operation of the ArmatureDisplay component in the script.

> - For DragonBones related scripting interface, please refer to the [DragonBones API](__APIDOC__/zh/modules/dragonBones.html).

Reviewer: ```suggestion
- For DragonBones ArmatureDisplay related scripting interface, please refer to the [ArmatureDisplay API](__APIDOC__/zh/classes/dragonbones.armaturedisplay-1.html).
```

Reviewer 2: nice catch!

> - For DragonBones ArmatureDisplay related scripting interface, please refer to the [ArmatureDisplay API](__APIDOC__/zh/classes/dragonbones.armaturedisplay-1.html).

Reviewer: ```suggestion
- For DragonBones ArmatureDisplay related scripting interface, please refer to the [ArmatureDisplay API](__APIDOC__/en/classes/dragonbones.armaturedisplay-1.html).
```

| Animation Cache Mode  | Rendering mode, default `REALTIME` mode. <br>1. **REALTIME** mode, real-time computing, supports all DragonBones features. <br>2. **SHARED_CACHE** mode, caches and shares skeletal animations and texture data, equivalent to pre-baked skeletal animations. Has higher performance, but does not support action fusion, action overlay, skeleton nesting, and only supports action start and end events. As for memory, it has a memory advantage when creating N(N>=3) animations with the same skeleton and the same action, the larger the N value, the more obvious the advantage. In summary, `SHARED_CACHE` mode is suitable for scene animations, effects, replica monsters, NPCs, etc., and can greatly improve frame rates and reduce memory consumption. <br>3. **PRIVATE_CACHE** mode, similar to `SHARED_CACHE` but does not share animation and texture data, so there is no memory advantage, only performance advantage. If you want to take advantage of the high performance of the cache mode, and also want to implement skins function (cannot share the texture data), then `PRIVATE_CACHE` is for you.
| Time Scale            | The time scale of all animations in the current skeleton

> | Play Times            | The number of cycles to play the default animation. <br>**-1** means use the default value in the configuration file;<br>**0** means infinite loop;<br>**>0** means loop times

Reviewer: ```suggestion
| Play Times            | The number of cycles to play the default animation. <br>**-1** means use the default value in the configuration file.<br>**0** means infinite loop.<br>**>0** means loop times.
```

| PremultipliedAlpha    | Whether the image is enabled for texture pre-multiplied, default is True.<br>This item needs to be turned off when the transparent area of the image appears color blocked. <br>Enabled when the translucent areas of the image become black.
| DebugBones            | Whether to show debug information for bone
| Sockets               | A attachment system for attaching certain external nodes to a given bone joint

> **Note**: the **Anchor** and **Size** properties on the Node component in the **Inspector** panel are disabled when using the ArmatureDisplay component.

## DragonBones Skins

The following is an example of how DragonBones implement skins function. Replace the weapon in the green box below with the knife in the red box by replacing the display object of the slot.

![dragonbones-cloth](./dragonbones/cloth.png)

> 1. First create a new empty node in the **Hierarchy** panel and rename it to replaceDBNode. then add ArmatureDisplay component in **Inspector** panel. And drag and drop the assets of the knife in the red box to the properties box of the ArmatureDisplay component, as follows:

Reviewer 2: replaceDBNode
-->
`replaceDBNode`
